rotkehlchen/chain/gnosis/modules/gnosis_pay/decoder.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any
+
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.gnosis.modules.gnosis_pay.constants import (
+    CPT_GNOSIS_PAY,
+    GNOSIS_PAY_CASHBACK_ADDRESS,
+    GNOSIS_PAY_CPT_DETAILS,
+)
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+
+
+class GnosisPayDecoder(DecoderInterface):
+
+    def decode_cashback_events(self, context: DecoderContext) -> DecodingOutput:
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
```

Comment:
Be more specific. The only cashback we can receive for now is GNO, so check for that asset. I mean in general this is kind of hacky anyway since we depend on Karpatkey's promise they made this safe to payout GNO cashback. So let's at least narrow down that promise.